rfcs/45-stageless.md

```diff
@@ -40,8 +40,6 @@ Let's define some terms so that hopefully we're all on the same page.
 - **schedule** (noun): the executable form of a system set
 - **executor**: executes a schedule on a world
 - **"ready"**: when a system is no longer waiting for dependencies to complete
-- **"complete"**: when a system has finished running or was skipped (because a condition didn't pass)

```

Comment:
“complete” is used by “dependency” and “ready”.

Reply:
Do you think we should revert this change then?

Reply:
I think so, otherwise we're back at the point that “complete” is undefined.

Reply:
agreed. 

I think we should move the glossary out of User Facing Description since some of these terms aren't used there.

Reply:
Terms used in the user facing description should stay there. Additional terms used in the implementation strategy have to be added there. This will also help to separate concerns between the two.

For example, it has been established that system sets and system labels are now merged concepts. However, in the implementation strategy the two seem to split again.

Reply:
Saying labels and sets have been "merged" is probably the wrong way to talk about it. Might be more accurate to say that for set and schedule.

The key thing is a system set is no longer a container. It's a label that is mapped to a dependency graph. This and the removal of stages decouple scheduling from storage.

Likewise, `.in_set` being a graph edge just like `.before` and `.after` (1) gives users and plugin authors a natural way to express multi-level dependency graphs and (2) makes scheduling completely order-independent; everything graph-related gets deferred until the app builds.

And the ultimate benefit is that these changes will make it easier to work with plugins.